src/pages/VenueLandingPage/VenueLandingPage.tsx

```diff
@@ -91,6 +104,56 @@ export const VenueLandingPage: React.FunctionComponent<VenueLandingPageProps> =
     }
   }, [shouldOpenPaymentModal, isAuthenticationModalOpen]);
 
+  const onJoinClick = useCallback(() => {
+    if (venueId) {
+      const venueEntrance = venue?.entrance && venue.entrance.length;
+      window.location.href =
+        user && !venueEntrance
+          ? venueInsideUrl(venueId)
+          : venueEntranceUrl(venueId);
+    }
+  }, [user, venue, venueId]);
```

Comment:
We tend to use the 'guard' pattern instead of wrapping the entire block within an `if` statement:

```suggestion
  const onJoinClick = useCallback(() => {
    if (!venueId) return
    
    const venueEntrance = venue?.entrance && venue.entrance.length;
    
    window.location.href =
      user && !venueEntrance
        ? venueInsideUrl(venueId)
        : venueEntranceUrl(venueId);
  }, [user, venue, venueId]);
```

Also, similar comments on `venueEntrace` that I made elsewhere in this review.